core/base/inc/DllImport.h

```diff
@@ -29,5 +29,12 @@
 # define R__EXTERN extern
 #endif
 
+#ifndef R__DLLEXPORT
+# ifdef _MSC_VER
+#  define R__DLLEXPORT __declspec(dllexport)
```

Comment:
What about `__declspec(dllexport)` - don't we need this usual "if we are building the library" vs "if we are linking the library", as in https://docs.microsoft.com/en-us/cpp/build/importing-into-an-application-using-declspec-dllimport?view=vs-2019 ?

Reply:
You mean `__declspec(dllexport)` I suppose? Not sure, let's try...

Reply:
@Axel-Naumann So we have two use cases here:
 1) we need to find the symbols from the interpreter (e.g. `.x box.cxx`)
 2) we need to import the symbols when linking (e.g. when creating a stand-alone app)

The provided solution works for both